plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlScannerTest.java

```diff
@@ -17,20 +17,64 @@
 package com.google.cloud.tools.eclipse.appengine.validation;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jdt.core.IJavaProject;
 import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
 import org.junit.Test;
 import org.xml.sax.SAXException;
 import org.xml.sax.ext.Locator2Impl;
 import org.xml.sax.helpers.AttributesImpl;
 
+import com.google.cloud.tools.eclipse.test.util.project.TestProjectCreator;
+
 public class WebXmlScannerTest {
 
-  private WebXmlScanner scanner = new WebXmlScanner();
+  private static IFile resource;
+  private static IJavaProject javaProject;
+  private WebXmlScanner scanner;
+  
+  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator();
```

Comment:
With the recent change to `TestProjectCreator`, you must explicitly provide the JavaFacet, and I think you'll also want the Dynamic Web Facet:
```
new TestProjectCreator().withFacetVersions(JavaFacet.VERSION_1_7, WebFacetUtils.WEB_25);
```
Note that the Dynamic Web Facet puts `web.xml` under `WebContent/` rather than `src/main/webapp` (which is Maven-ish).